components/server/src/api/envvar-service-api.ts

```diff
@@ -0,0 +1,178 @@
+        }
+
+        return response;
+    }
+
+    async createEnvironmentVariable(
+        req: CreateEnvironmentVariableRequest,
+        context: HandlerContext,
+    ): Promise<CreateEnvironmentVariableResponse> {
+        const response = new UpdateEnvironmentVariableResponse();
+        if (req.scope.case === "userScope") {
+            const variable: UserEnvVarValue = {
+                name: req.envVarName || "",
+                value: req.envVarValue || "",
+                repositoryPattern: req.scope.value.repositoryPattern || "",
+            };
+            variable.repositoryPattern = UserEnvVar.normalizeRepoPattern(variable.repositoryPattern);
+
+            await this.envVarService.addUserEnvVar(context.user.id, context.user.id, variable);
+
+            const updatedUserEnvVars = await this.envVarService.listUserEnvVars(context.user.id, context.user.id);
+            const updatedUserEnvVar = updatedUserEnvVars.find(
+                (v) => v.name === req.envVarName && v.repositoryPattern === variable.repositoryPattern,
+            );
+            if (!updatedUserEnvVar) {
+                throw new ConnectError("env variable not found", Code.Internal);
+            }
+
+            response.environmentVariable = this.apiConverter.toEnvironmentVariable(updatedUserEnvVar);
+        } else if (req.scope.case === "configurationScope") {
+            const variable: ProjectEnvVarWithValue = {
+                name: req.envVarName || "",
+                value: req.envVarValue || "",
+                censored: req.scope.value.censored || false,
+            };
+
+            await this.envVarService.addProjectEnvVar(context.user.id, req.scope.value.configurationId, variable);
+
+            const updatedProjectEnvVars = await this.envVarService.listProjectEnvVars(
+                context.user.id,
+                req.scope.value.configurationId,
+            );
+            const updatedProjectEnvVar = updatedProjectEnvVars.find((v) => v.name === req.envVarName);
+            if (!updatedProjectEnvVar) {
+                throw new ConnectError("env variable not found", Code.Internal);
+            }
+
+            response.environmentVariable = this.apiConverter.toEnvironmentVariable(updatedProjectEnvVar);
+        } else {
+            throw new ConnectError("scope is not set", Code.InvalidArgument);
+        }
+
+        return response;
+    }
+
+    async deleteEnvironmentVariable(
+        req: DeleteEnvironmentVariableRequest,
+        context: HandlerContext,
+    ): Promise<DeleteEnvironmentVariableResponse> {
+        const variable: UserEnvVarValue = {
+            id: req.envVarId,
+            name: "",
+            value: "",
+            repositoryPattern: "",
+        };
+        await this.envVarService.deleteUserEnvVar(context.user.id, context.user.id, variable);
+        try {
+            await this.envVarService.deleteProjectEnvVar(context.user.id, req.envVarId);
+        } catch (e) {
```

Comment:
Not sure about thiss as there's no way to know if it's user or project env just with the id